restcomm.android.client.sdk/src/main/java/org/mobicents/restcomm/android/sipua/impl/SipManager.java

```diff
@@ -878,6 +882,11 @@ else if (response.getStatusCode() == Response.RINGING) {
 			//RCLogger.i(TAG, "BUSY");
 			dispatchSipEvent(new SipEvent(this,
 					SipEventType.SERVICE_UNAVAILABLE, "", ""));
+		} else if (response.getStatusCode() == Response.FORBIDDEN) {
```

Comment:
@mehturt: good call 👍. However keep in mind that FORBIDDEN could be returned in all of REGISTER, INVITE, MESSAGE requests, so for better error handling I would suggest returning:
```
dispatchSipError(ISipEventListener.ErrorContext.ERROR_CONTEXT_NON_CALL,RCClient.ErrorCodes.SIGNALLING_REGISTER_AUTH_ERROR, "Error authenticating " + tid.getRequest().getMethod());
```
for REGISTER/MESSAGE requests, BUT for INVITE requests instead return:

```
dispatchSipError(ISipEventListener.ErrorContext.ERROR_CONTEXT_CALL, RCClient.ErrorCodes.SIGNALLING_REGISTER_AUTH_ERROR, "Error authenticating " + tid.getRequest().getMethod());
```

Which differentiates between ERROR_CONTEXT_CALL and ERROR_CONTEXT_NON_CALL